meta/SearchResult.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace dokuwiki\plugin\struct\meta;
+
+/**
+ * Class SearchResult
+ *
+ * Search is executed only once per request.
+ */
+class SearchResult
+{
+    /** @var Value[][] */
+    protected $rows = [];
+    /** @var array */
+    protected $pids = [];
+    protected $rids = [];
+    /** @var array */
+    protected $revs = [];
+    /** @var int */
+    protected $count = -1;
+
+    /**
+     * Construct SearchResult
+     *
+     * @param \PDOStatemen $res
+     * @param int $rangeBegin
+     * @param int $rangeEnd
+     * @param array $columns
+     * @param bool $pageidAndRevOnly
```

Comment:
can we have some parameter descriptions? especially the last one is a bit mysterious ;-)